scripts/publish.js

```diff
@@ -0,0 +1,53 @@
+const execSync = require("child_process").execSync;
+
+process.on('unhandledRejection', err => {
+  throw err;
+});
+
+const exec = (command, extraEnv) =>
+  execSync(command, {
+    stdio: "inherit",
+    env: Object.assign({}, process.env, extraEnv)
+  });
+
+const execOut = (command, extraEnv) =>
+  execSync(command, {
+    stdio: "pipe",
+    env: Object.assign({}, process.env, extraEnv)
+  }).toString('utf8').trim();
+
+const branch = execOut('git rev-parse --symbolic-full-name --abbrev-ref HEAD');
+
+if (branch !== 'master') {
+  console.log('You must be on branch "master" to run: npm run publish');
+  process.exit(1);
+}
+
+const bump = process.argv[3];
+
+if (!bump || !['major', 'minor', 'patch'].includes(bump)) {
+  console.log(`Please pass (major|minor|patch) as an argument.`);
+  console.log(`example: npm run publish minor`);
+  process.exit(1);
+}
+
+console.log('Making sure we have the latest master changes.');
+exec('git pull');
+
+console.log(`Creating new ${bump} version of react-governor.`);
+exec('npm run build');
+
+// This will fail and exit if the git working directory is not clean
+// This updates and commits the package.json
+console.log('Updating package.json');
+exec(`npm version ${bump}`);
+
+const packageJSON = require("../package.json");
+console.log('Pushing to npm repository');
+exec('npm publish --access public');
+
+console.log('Tagging branch');
+exec(`git tag v${packageJSON.version}`);
+
+console.log('Pushing to master');
+exec('git push');
```

Comment:
1. Won't this throw an error if you haven't set an upstream for your branch?
2. How do tags work across forks? If a push a tag to my fork, can I open a pull request to mirror that (I don't think so)?

Reply:
We're never publishing from a branch other than master. All feature branches would be reviewed, a pull request would be made, once it's merged into master, then you publish from master.

Reply:
That's not my concern. My concern is that now I can only publish from a clone of this repository on master, while I do all my work on my fork.